projects/admin/src/app/record/detail-view/patron-detail-view/patron-detail-view.component.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Component, OnInit } from '@angular/core';
+import { DetailRecord } from '@rero/ng-core/lib/record/detail/view/detail-record';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'admin-patron-detail-view',
+  templateUrl: './patron-detail-view.component.html'
+})
+export class PatronDetailViewComponent implements OnInit, DetailRecord {
+
+  record$: Observable<any>;
+
+  record: any;
+
+  type: string;
+
+  ngOnInit() {
+    this.record$.subscribe(record => this.record = record);
+  }
+
+  get isPatron() {
+    if (this.record && this.record.metadata.roles) {
+      return this.record.metadata.roles.some((role: string) => role === 'patron');
+    }
+    return false;
+  }
+
+  get isLibrarian() {
+    if (this.record && this.record.metadata.roles) {
+      return this.record.metadata.roles.some((role: string) => role === 'librarian');
+    }
+    return false;
+  }
```

Comment:
You already defined the function `isPatron()` into `projects/admin/src/app/record/brief-view/patrons-brief-view.component.ts` file. Is a a way to declare it once and reuse it (maybe part of a `patron.sevice.ts` file) ? (I'm not an Angular expert).

Reply:
done